File: specification/specs/x_pack/security/api_key/create_api_key/CreateApiKeyRequest.ts
@rest_spec_name("security.create_api_key")
class CreateApiKeyRequest extends RequestBase {
	query_parameters: {
		refresh: Refresh;
	}
	body: {
		expiration: Time;
		name: string;
		role_descriptors: Dictionary<string, ApiKeyRole>;
	}
}

File: output/csharp/XPack/Security/ApiKey/CreateApiKey/CreateApiKeyRequest.cs
using Nest.Common;
using Nest.CommonOptions;
using Nest.XPack;
using Nest.CommonAbstractions;
using System;
using System.Collections.Generic;
using System.Runtime.Serialization;
namespace Nest.XPack {

	public class CreateApiKeyRequest : RequestBase {
		
		[DataMember(Name="refresh")]
		public Refresh Refresh { get; set; }


		[DataMember(Name="expiration")]
		public Time Expiration { get; set; }


		[DataMember(Name="name")]
		public string Name { get; set; }


		[DataMember(Name="role_descriptors")]
		public IDictionary<string, ApiKeyRole> RoleDescriptors { get; set; }

	}
}
